draft-ietf-mls-protocol.md

```diff
@@ -1513,82 +1513,20 @@ struct {
     opaque group_id<0..255>;
     uint64 epoch;
     ContentType content_type;
+    opaque masked_sender_data<0..255>;
     opaque authenticated_data<0..2^32-1>;
-    opaque sender_data_nonce<0..255>;
-    opaque encrypted_sender_data<0..255>;
     opaque ciphertext<0..2^32-1>;
+    opaque auth_tag<0..255>;
 } MLSCiphertext;
 ~~~~~
 
 External sender types are sent as MLSPlaintext, see {{external-proposals}}
 for their use.
 
 The remainder of this section describes how to compute the signature of an
-MLSPlaintext object and how to convert it to an MLSCiphertext object for
-`member` sender types.  The steps are:
-
-* Set group_id, epoch, content_type and authenticated_data fields from the
-  MLSPlaintext object directly
-
-* Randomly generate the sender_data_nonce field
-
-* Identify the key and key generation depending on the content type
-
-* Encrypt an MLSSenderData object for the encrypted_sender_data field from
-  MLSPlaintext and the key generation
-
-* Generate and sign an MLSPlaintextTBS object from the MLSPlaintext
-  object
-
-* Encrypt an MLSCiphertextContent for the ciphertext field using the key
-  identified, the signature, and MLSPlaintext object
-
-Decryption is done by decrypting the metadata, then the message, and then
-verifying the content signature.
-
-The following sections describe the encryption and signing processes in detail.
-
-## Metadata Encryption
-
-The "sender data" used to look up the key for the content encryption
-is encrypted under AEAD using the MLSCiphertext sender_data_nonce and
-the sender_data_key from the keyschedule. It is encoded as an
-object of the following form:
-
-~~~~~
-struct {
-    uint32 sender;
-    uint32 generation;
-    opaque reuse_guard[4];
-} MLSSenderData;
-~~~~~
-
-MLSSenderData.sender is assumed to be a `member` sender type.  When constructing
-an MLSSenderData from a Sender object, the sender MUST verify Sender.sender_type
-is `member` and use Sender.sender for MLSSenderData.sender.
-
-The `reuse_guard` field contains a fresh random value used to avoid nonce reuse
-in the case of state loss or corruption, as described in
-{{content-signing-and-encryption}}.
-
-The Additional Authenticated Data (AAD) for the SenderData ciphertext
-computation is its prefix in the MLSCiphertext, namely:
-
-~~~~~
-struct {
-    opaque group_id<0..255>;
-    uint64 epoch;
-    ContentType content_type;
-    opaque authenticated_data<0..2^32-1>;
-} MLSSenderDataAAD;
-~~~~~
-
-When parsing a SenderData struct as part of message decryption, the
-recipient MUST verify that the sender field represents an occupied
-leaf in the ratchet tree.  In particular, the sender index value
-MUST be less than the number of leaves in the tree.
+MLSPlaintext object and how to convert it to an MLSCiphertext object.
```

Comment:
I think the overview steps are important to have here, especially since the ciphertext sampling makes the order of things a bit complex (you have to encrypt the content, then loop back to the sender data).  But I think we streamline a little bit:

```
* Set group_id, epoch, content_type and authenticated_data fields from the
  MLSPlaintext object directly

* Identify the key and key generation depending on the content type

* Encrypt an MLSCiphertextContent for the ciphertext field using the key
  identified and MLSPlaintext object

* Encrypt the sender data using the sender data key for the epoch and a
   nonce sampled from  encrypted MLSCiphertextContent.
```